src/err/mod.rs

```diff
@@ -538,12 +538,25 @@ impl PyErr {
     where
         T: ToPyObject,
     {
-        self.is_instance(py, exc.to_object(py).as_ref(py))
+        self.is_instance_bound(py, exc.to_object(py).bind(py).clone())
     }
 
-    /// Returns true if the current exception is instance of `T`.
+    /// Deprecated form of `PyErr::is_instance_bound`.
+    #[cfg_attr(
+        not(feature = "gil-refs"),
+        deprecated(
+            since = "0.21.0",
+            note = "`PyErr::is_instance` will be replaced by `PyErr::is_instance_bound` in a future PyO3 version"
+        )
+    )]
     #[inline]
     pub fn is_instance(&self, py: Python<'_>, ty: &PyAny) -> bool {
+        self.is_instance_bound(py, ty.into_py(py).into_bound(py))
+    }
+
+    /// Returns true if the current exception is instance of `T`.
+    #[inline]
+    pub fn is_instance_bound(&self, py: Python<'_>, ty: Bound<'_, PyAny>) -> bool {
```

Comment:
Let's use `&Bound` here as the input, as we don't need to take ownership of the type:

```suggestion
    pub fn is_instance_bound(&self, py: Python<'_>, ty: &Bound<'_, PyAny>) -> bool {
```